src/DependencyInjection/Compiler/AddRenderersPass.php

```diff
@@ -36,16 +36,10 @@ public function process(ContainerBuilder $container)
             }
         }
 
-        if (class_exists(ServiceLocatorTagPass::class)) {
-            $locator = ServiceLocatorTagPass::register($container, $rendererReferences);
-            // Replace the service definition with a PsrProvider
-            $container->register('knp_menu.renderer_provider', PsrProvider::class)
-                ->addArgument($locator)
-                ->addArgument('%knp_menu.default_renderer%');
-        } else {
-            // BC for Symfony < 3.3
-            $definition = $container->getDefinition('knp_menu.renderer_provider');
-            $definition->replaceArgument(2, $renderers);
-        }
+        $locator = ServiceLocatorTagPass::register($container, $rendererReferences);
+        // Replace the service definition with a PsrProvider
```

Comment:
the replacement itself should be dropped. The service should be a PsrProvider directly in the XML config file now, and we should only replace the first argument with the locator here.

Reply:
@stof in a previous comment you said that definition of `Knp\Menu\Provider\PsrProvider` service should be dropped (since it's removed using LazyProvider), now you say that the service should be in XML config. Can you clarify?

Reply:
you are confusing things. This compiler pas is about the renderer provider, not the menu provider.

The PsrProvider for renderer provider must be used, as that's what get used for Symfony 3.3+. But the service definition using the PsrProvider should be in the XML file directly, instead of defining the service for the old ContainerAwareProvider in XML and replacing it entirely here.
Then, this compiler pass should be only replacing the argument to inject the service locator.

Reply:
So, should we remove ContainerAwareProvider class to?